core/src/banking_stage.rs

```diff
@@ -440,20 +689,40 @@ impl BankingStage {
         let data_budget = Arc::new(DataBudget::default());
         let batch_limit =
             TOTAL_BUFFERED_PACKETS / ((num_threads - NUM_VOTE_PROCESSING_THREADS) as usize);
+        // Keeps track of extraneous vote transactions for the vote threads
+        let latest_unprocessed_votes = Arc::new(LatestUnprocessedVotes::new());
+        let bank = poh_recorder.read().unwrap().bank();
+        // TODO: add a secondary feature flag and rework this to be able to hot swap into the new
+        // data structure
+        let split_voting_threads = bank
+            .map(|bank| {
+                bank.feature_set
+                    .is_active(&allow_votes_to_directly_update_vote_state::id())
+            })
+            .unwrap_or(false);
         // Many banks that process transactions in parallel.
         let bank_thread_hdls: Vec<JoinHandle<()>> = (0..num_threads)
             .map(|i| {
-                let (verified_receiver, forward_option) = match i {
+                let (verified_receiver, transaction_storage_options) = match i {
```

Comment:
the issue i had here is that the voting threads should have the same reference to `latest_unprocessed_votes` while the banking threads should each make their own `UnprocessedTransactionStorage` so that they have a local `UnprocessedPacketBatches`. I could make a `clone` function that copies the reference for the `VoteStorage` struct and creates a new `UnprocessedPacketBatches` for the `TransactionStorage` struct.